airflow/providers/google/cloud/operators/dataflow.py

```diff
@@ -677,6 +677,9 @@ def set_current_job(current_job):
         options = self.dataflow_default_options
         options.update(self.options)
 
+        if not self.location:
```

Comment:
why can't we change this in the init method?

Reply:
because we need to store the original state for location in the class (in case if location parameter was missed)

Reply:
> why can't we change this in the init method?
because we need to store the original state for location in the class (in case if location parameter was missed)